refactor/data/openapi.yaml

```diff
@@ -0,0 +1,344 @@
+      description: Get a result object
+      operationId: get_result_api_v0_workflow_results__dispatch_id__get
+      parameters:
+        - required: true
+          schema:
+            title: Dispatch Id
+            type: string
+          name: dispatch_id
+          in: path
+      responses:
+        '200':
+          description: Successful Response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResultPickle'
+        '404':
+          description: The result was not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/HTTPExceptionSchema'
+        '422':
+          description: Validation Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/HTTPValidationError'
+    put:
+      tags:
+        - Workflow
+      summary: Update Result
+      description: Update a result object
+      operationId: update_result_api_v0_workflow_results__dispatch_id__put
+      parameters:
+        - required: true
+          schema:
+            title: Dispatch Id
+            type: string
+          name: dispatch_id
+          in: path
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Node'
+        required: true
+      responses:
+        '200':
+          description: Successful Response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/UpdateResultResponse'
+        '422':
+          description: Validation Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/HTTPValidationError'
+  /api/v0/workflow/results:
+    post:
+      tags:
+        - Workflow
+      summary: Insert Result
+      description: Insert a result object
+      operationId: insert_result_api_v0_workflow_results_post
+      requestBody:
+        content:
+          application/json:
+            schema:
+              title: Result Object
+              anyOf:
+                - $ref: '#/components/schemas/Result'
+                - $ref: '#/components/schemas/ResultPickle'
+        required: true
+      responses:
+        '200':
+          description: Successful Response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/InsertResultResponse'
+        '422':
+          description: Validation Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/HTTPValidationError'
+  /api/v0/fs/upload:
+    post:
+      tags:
+        - Data
+      summary: Upload File
+      description: Upload a file
+      operationId: upload_file_api_v0_fs_upload_post
+      requestBody:
+        content:
+          multipart/form-data:
```

Comment:
Is this the right `POST` data type for the server to pass a `FileStorage` object to the backend? I think this is the type we want to use to wrap files https://werkzeug.palletsprojects.com/en/2.0.x/datastructures/#werkzeug.datastructures.FileStorage

Reply:
I think the intended use case of this endpoint is something like:

```
import requests

files = {'file': ('foo.txt', open('./foo.txt', 'rb'))}
response = requests.post('http://127.0.0.1:8000/api/v0/fs/upload', files=files)
print(response)
print(response.json())
```
the fast API server
```
from fastapi import FastAPI, File, UploadFile
import uvicorn

app = FastAPI()

@app.post('/file')
def _file_upload(my_file: UploadFile = File(...)):
    print(my_file)

if __name__ == "__main__":
    uvicorn.run("main:app", host="0.0.0.0", port=8000, log_level="debug")
```

source: https://stackoverflow.com/questions/65102579/send-and-receive-file-using-python-fastapi-and-requests

Reply:
I did a little googling on this, and I think you're right that `multipart/form-data` is the correct content type. Although I'm not sure about the `_file_upload` implementation above, I think there might be some issues with that, but for the skeleton I think it's OK.